packages/react/src/__tests__/ReactContextValidator-test.js

```diff
@@ -431,8 +431,12 @@ describe('ReactContextValidator', () => {
     expect(renderContext).toBe(secondContext);
     expect(componentDidUpdateContext).toBe(secondContext);
 
-    // sCU is not called in this case because React force updates when a provider re-renders
-    expect(shouldComponentUpdateWasCalled).toBe(false);
+    if (gate(flags => flags.enableLazyContextPropagation)) {
+      expect(shouldComponentUpdateWasCalled).toBe(true);
```

Comment:
Consider adding a comment similar to the other branch so that when that branch is deleted there's an explanation of the expectation here. Something like:

```suggestion
      // sCU is called in this case because React does not force updates when a provider re-renders.
      // Instead, we check context changes lazily, after sCU is called.
      expect(shouldComponentUpdateWasCalled).toBe(true);
```